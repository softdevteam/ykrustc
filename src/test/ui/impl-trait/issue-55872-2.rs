// edition:2018
// ignore-compare-mode-chalk

// revisions: min_tait full_tait
#![feature(min_type_alias_impl_trait)]
#![cfg_attr(full_tait, feature(type_alias_impl_trait))]
//[full_tait]~^ WARN incomplete

pub trait Bar {
    type E: Copy;

    fn foo<T>() -> Self::E;
}

impl<S> Bar for S {
    type E = impl std::marker::Copy;
    //~^ ERROR the trait bound `impl Future: Copy` is not satisfied [E0277]
    fn foo<T>() -> Self::E {
        //~^ ERROR type parameter `T` is part of concrete type but not used in parameter list for the `impl Trait` type alias
        async {}
    }
}

fn main() {}
