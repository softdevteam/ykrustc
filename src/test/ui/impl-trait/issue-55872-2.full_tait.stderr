warning: the feature `type_alias_impl_trait` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/issue-55872-2.rs:6:32
   |
LL | #![cfg_attr(full_tait, feature(type_alias_impl_trait))]
   |                                ^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information

error[E0277]: the trait bound `impl Future: Copy` is not satisfied
  --> $DIR/issue-55872-2.rs:16:14
   |
LL |     type E = impl std::marker::Copy;
   |              ^^^^^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `impl Future`

error: type parameter `T` is part of concrete type but not used in parameter list for the `impl Trait` type alias
  --> $DIR/issue-55872-2.rs:18:28
   |
LL |       fn foo<T>() -> Self::E {
   |  ____________________________^
LL | |
LL | |         async {}
LL | |     }
   | |_____^

error: aborting due to 2 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0277`.
